### PR TITLE
Low coverage warning

### PR DESCRIPTION
**Pull request by agnepagne**
Introduces text based on number of publications and WoS coverage in abm.Rmd.

Also slightly changed WoS attribution and inclusion of this in xlsx export.

### CODE REVIEW

**Comment by agnepagne** [comment]
Guess it's best to have @mskyttner reviewing since @agnepagne and @fileunderwater worked on the branch.

Don't pay too much attention to the wos_disclaimer() which (as opposed to wos_attribution()) is not really in use.

File: inst/extdata/abm.Rmd [hunk 1]
```diff
@@ -208,10 +208,15 @@ if(wospubs < 10)
 wospubstext <- paste0(wospubstext, "</p>")
 cat(wospubstext)
 
-coveragetext <- paste0("<p>Web of Science coverage is ", 100*round(totcov, 3), "%. ")
-if(totcov >= 0.8) coveragetext <- paste0(coveragetext, "This is OK!</p>")
-if(totcov < 0.8 && totcov >= 0.6) coveragetext <- paste0(coveragetext, "<b>This is bad, proceed with caution!</b></p>")
-if(totcov < 0.6) coveragetext <- paste0(coveragetext, "<b>This is very bad, indicators can not be trusted!</b></p>")
+coveragejudge <- ifelse(totcov >= 0.8, "good",
+                        ifelse(totcov >= 0.65, "moderate",
+                               ifelse(totcov >= 0.5, "poor", "bad")))
```
Comment (mskyttner) [inline inst/extdata/abm.Rmd]: compact way that works, an alternative is to use https://adv-r.hadley.nz/control-flow.html#switch here

File: DESCRIPTION [hunk 2]
```diff
@@ -49,9 +49,9 @@ Imports:
     scales (>= 1.0.0),
     glue (>= 1.3.1),
     rmarkdown (>= 1.16),
-    base64enc (>= 0.1-3),
-    RColorBrewer,
-    openssl
+    base64enc (>= 0.1.3),
```
Comment (mskyttner) [inline DESCRIPTION]: good catch!

File: R/abm.R [hunk 3]
```diff
@@ -604,7 +651,8 @@ abm_private_data <- function(unit_code) {
       abm_table3(con = db, unit_code = x),
       abm_table4(con = db, unit_code = x),
       abm_table5(con = db, unit_code = x),
-      publications = abm_publications(con = db, unit_code = x)
+      abm_woscoverage(con = db, unit_code = x),
```
Comment (mskyttner) [inline R/abm.R]: this list slot was named in abm_public_data() but not here? ie `coverage = abm_woscoverage(...)`

File: inst/extdata/abm.Rmd [hunk 4]
```diff
@@ -133,7 +138,6 @@ if(df_cf %>% filter(!is.na(P_frac)) %>% nrow() > 0){
   
   grid.arrange(pub_bullet1, pub_bullet2, ncol = 1)
```
Comment (mskyttner) [inline inst/extdata/abm.Rmd]: patchwork? https://www.datanovia.com/en/blog/ggplot-multiple-plots-made-ridiculuous-simple-using-patchwork-r-package/

File: R/abm.R (hunk 3 above)
Comment (agnepagne) [reply R/abm.R]: Good catch. Fixing.

File: inst/extdata/abm.Rmd [hunk 5]
```diff
@@ -195,6 +196,27 @@ if(df_copub %>% filter(!is.na(P_full)) %>% nrow > 0){
 Row
 --------------------------------
 
+### **Notes**
+```{r, echo=FALSE, results="asis"}
+
+if (params$is_employee == TRUE)
+  cat("<p><b>Bibliometric results for individual researchers should always be treated with caution.</b></p>")
```
[the remark this answers is not in the document]
Comment (agnepagne) [reply inst/extdata/abm.Rmd]: Probably better, yes.

File: inst/extdata/abm.Rmd (hunk 1 above)
Comment (agnepagne) [reply inst/extdata/abm.Rmd]: Fetching from function in latest version, so not a current concern. Does switch work with intervals?

File: inst/extdata/abm.Rmd (hunk 4 above)
Comment (agnepagne) [reply inst/extdata/abm.Rmd]: Very interesting package, thanks for link!

File: inst/extdata/abm.Rmd (hunk 1 above)
Comment (mskyttner) [reply inst/extdata/abm.Rmd]: This SO post seems to say ifelse may be preferred if not `cut` is even better? 

https://stackoverflow.com/questions/30293465/in-r-how-to-interpret-switch-statement-with-greater-than-less-than

```r
totcov <- 0.3
as.character(cut(totcov, right = F,
 c(0, 0.5, 0.65, 0.8, 1), 
 c("bad", "poor", "moderate", "good")))
```

Or for reuse:

```r
cvrg <- runif(10)  # one or several

judge <- function(x) {
  if (is.na(x) || is.null(x)) return (NA)
  as.character(cut(x, right = FALSE,
    breaks = c(0, 0.5, 0.65, 0.8, 1),
    labels = c("bad", "poor", "moderate", "good")
  ))
}

judge(cvrg)
judge(cvrg[1])
judge(NA)
judge(NULL)
```

Comment (mskyttner) [reply inst/extdata/abm.Rmd]: it already works well, right?

File: inst/extdata/abm.Rmd (hunk 4 above)
Comment (mskyttner) [reply inst/extdata/abm.Rmd]: this was mostly an idea for future refactoring...

**Pull request merged**